src/backend/joanie/core/api/client.py

```diff
@@ -281,12 +281,17 @@ class OrderViewSet(
 
     def get_queryset(self):
         """Custom queryset to limit to orders owned by the logged-in user."""
-        username = (
-            self.request.auth["username"]
-            if self.request.auth
-            else self.request.user.username
-        )
-        return models.Order.objects.filter(owner__username=username).select_related(
+        query_filter = Q()
+        if hasattr(self.request.user, "get_abilities"):
+            abilities = self.request.user.get_abilities(self.request.user)
+            if not abilities["has_organization_access"]:
+                username = (
+                    self.request.auth["username"]
+                    if self.request.auth
+                    else self.request.user.username
+                )
+                query_filter = Q(owner__username=username)
+        return models.Order.objects.filter(query_filter).select_related(
```

Comment:
We can't retrieve user organization order in this manner as this viewset if used by learner to retrieve their order. That means a user that is both a learner and organization member will retrieve its orders and organization orders in the response payload